semgrep/semgrep/core_runner.py

```diff
@@ -199,99 +199,98 @@ def _run_rule(
-                    ]
+                debug_print(core_run.stderr.decode("utf-8", "replace"))
 
-                    if equivalences:
-                        cmd += ["-equivalences", equiv_fout.name]
-                    cmd += ["-j", str(self._jobs)]
-                    cmd += [str(path) for path in targets]
-
-                    debug_print(f"Running semgrep... '{cmd}'")
-                    core_run = subprocess.run(
-                        cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-                    )
-
-                    debug_print(core_run.stderr.decode("utf-8", "replace"))
-
-                    if core_run.returncode != 0:
-                        try:
-                            # see if semgrep output a JSON error that we can decode
-                            semgrep_output = core_run.stdout.decode("utf-8", "replace")
-                            output_json = json.loads(semgrep_output)
-                            if "error" in output_json:
-                                self._semgrep_error_json_to_message_then_exit(
-                                    output_json
-                                )
-                            else:
-                                raise SemgrepError(
-                                    f"unexpected non-json output while invoking semgrep-core:\n{PLEASE_FILE_ISSUE_TEXT}"
-                                )
-                        except Exception as e:
+                if core_run.returncode != 0:
+                    try:
+                        # see if semgrep output a JSON error that we can decode
+                        semgrep_output = core_run.stdout.decode("utf-8", "replace")
+                        output_json = json.loads(semgrep_output)
+                        if "error" in output_json:
+                            self._semgrep_error_json_to_message_then_exit(output_json)
+                        else:
                             raise SemgrepError(
-                                f"non-zero return code while invoking semgrep-core:\n\t{e}\n{PLEASE_FILE_ISSUE_TEXT}"
+                                f"unexpected non-json output while invoking semgrep-core:\n{PLEASE_FILE_ISSUE_TEXT}"
                             )
+                    except Exception as e:
+                        raise SemgrepError(
+                            f"non-zero return code while invoking semgrep-core:\n\t{e}\n{PLEASE_FILE_ISSUE_TEXT}"
+                        )
 
-                    output_json = json.loads(
-                        (core_run.stdout.decode("utf-8", "replace"))
-                    )
-                    errors.extend(output_json["errors"])
-                    outputs.extend([PatternMatch(m) for m in output_json["matches"]])
+                output_json = json.loads((core_run.stdout.decode("utf-8", "replace")))
+                errors.extend(output_json["errors"])
+                outputs.extend([PatternMatch(m) for m in output_json["matches"]])
```

Comment:
We can do `outputs.extend(PatternMatch(m) for m in output_json["matches"])` (no list comp) here to be a bit more memory efficient.